### PR TITLE
Scoped enums

### PR DESCRIPTION
**Pull request by TheMagnetar**
**When merged this pull request will:**
- Change some enums to scoped enums.
- Fixes one integer type overflow.

### CODE REVIEW

File: extensions/src/ACRE2Core/SoundEngine.cpp [hunk 1]
```diff
@@ -116,5 +116,5 @@ acre_result_t CSoundEngine::onEditCapturedVoiceDataEvent(short* samples, int sam
         }
     }
     */
-    return acre_result_ok;
+    return AcreResult::ok;
 }
```
Comment (jonpas) [inline extensions/src/ACRE2Core/SoundEngine.cpp]:
```suggestion
}

```

File: extensions/src/ACRE2Core/Self.h [hunk 2]
```diff
@@ -12,10 +12,10 @@ class CSelf : public CPlayer {
...
         this->setSpeaking(false);
-        this->setCurveModel(acre_curveModel_original);
+        this->setCurveModel(AcreCurveModel::original);
         this->setCurrentLanguageId(0);
     };
-    DECLARE_MEMBER(acre_curveModel_t, CurveModel);
+    DECLARE_MEMBER(AcreCurveModel, CurveModel);
     DECLARE_MEMBER(BOOL, Speaking);
     DECLARE_MEMBER(int, CurrentLanguageId);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/Self.h:21]:
```suggestion
};

```

File: extensions/src/ACRE2Core/RpcEngine.h [hunk 3]
```diff
@@ -14,14 +14,14 @@ class CRpcEngine : public TEntrantWorker<ACRE_RPCDATA *>
...
-    acre_result_t        runProcedure(IServer *serverInstance, IMessage *msg);
-    acre_result_t        runProcedure(IServer *serverInstance, IMessage *msg, BOOL entrant);
+    AcreResult        runProcedure(IServer *serverInstance, IMessage *msg);
+    AcreResult        runProcedure(IServer *serverInstance, IMessage *msg, BOOL entrant);
 
-    acre_result_t        exProcessItem(ACRE_RPCDATA *data);
+    AcreResult        exProcessItem(ACRE_RPCDATA *data);
 private:
     std::map<std::string, IRpcFunction *> m_FunctionList;
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/RpcEngine.h:27]:
```suggestion
};

```

File: extensions/src/ACRE2Core/KeyHandlerEngine.cpp [hunk 4]
```diff
@@ -102,24 +102,24 @@ acre_result_t CKeyHandlerEngine::setKeyBind(string eventName, int keyCode, BOOL
...
 
-acre_result_t CKeyHandlerEngine::clearKeybinds( void ) {
+AcreResult CKeyHandlerEngine::clearKeybinds( void ) {
     this->lock();
     this->m_keyMap.clear();
     this->unlock();
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 */
```
Comment (jonpas) [inline extensions/src/ACRE2Core/KeyHandlerEngine.cpp:125]:
```suggestion
*/

```

File: extensions/src/ACRE2Core/ext_reset.h [hunk 5]
```diff
@@ -16,16 +16,16 @@ RPC_FUNCTION(ext_reset) {
...
 
     // We should be calling this, but the named pipe function already calls this too
     // We leave this here so SQF can force a full unmute, though.
     CEngine::getInstance()->getClient()->unMuteAll();
 
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 DECLARE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/ext_reset.h:31]:
```suggestion
};

```

File: extensions/src/ACRE2Core/getClientID.h [hunk 6]
```diff
@@ -29,7 +29,7 @@ RPC_FUNCTION(getClientID) {
 
     UNLOCK(CEngine::getInstance()->getSelf());
     
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 DECLARE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/getClientID.h:35]:
```suggestion
};

```

File: extensions/src/ACRE2Core/getPluginVersion.h [hunk 7]
```diff
@@ -7,7 +7,7 @@ RPC_FUNCTION(getPluginVersion) {
 
     vServer->sendMessage(CTextMessage::formatNewMessage("handleGetPluginVersion", "%s", ACRE_VERSION));
 
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 DECLARE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/getPluginVersion.h:13]:
```suggestion
};

```

File: extensions/src/ACRE2Core/loadSound.h [hunk 8]
```diff
@@ -27,15 +27,15 @@ RPC_FUNCTION(loadSound) {
...
         } else {
             vServer->sendMessage(CTextMessage::formatNewMessage("handleLoadedSound", "%s,%s", id.c_str(), "0"));
         }
     }
 
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 DECLARE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/loadSound.h:41]:
```suggestion
};

```

File: extensions/src/ACRE2Core/localMute.h [hunk 9]
```diff
@@ -20,14 +20,14 @@ RPC_FUNCTION(localMute) {
...
     } else {
         CEngine::getInstance()->getClient()->enableMicrophone(true);
     }
 
 
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 DECLARE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/localMute.h:33]:
```suggestion
};

```

File: extensions/src/ACRE2Core/ping.h [hunk 10]
```diff
@@ -12,7 +12,7 @@ volatile DWORD g_pingTime;
 RPC_FUNCTION(ping) {
     g_pingTime = clock() / CLOCKS_PER_SEC;
     vServer->sendMessage(CTextMessage::formatNewMessage("pong", "%f,", g_pingTime));
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 DECLARE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/ping.h:18]:
```suggestion
};

```

File: extensions/src/ACRE2Core/playSound.h [hunk 11]
```diff
@@ -33,12 +33,12 @@ RPC_FUNCTION(playLoadedSound) {
...
+    AcreResult res = CEngine::getInstance()->getSoundPlayback()->playSound(id, position, direction, volume, isWorld == 1);
+    if (res == AcreResult::error) {
         vServer->sendMessage(CTextMessage::formatNewMessage("handleSoundError", "%s,", id.c_str()));
     }
 
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 DECLARE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/playSound.h:44]:
```suggestion
};

```

File: extensions/src/ACRE2Core/setMuted.h [hunk 12]
```diff
@@ -28,7 +28,7 @@ RPC_FUNCTION(setMuted) {
         CEngine::getInstance()->getClient()->setMuted(id, (BOOL)status);
     }
 
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 DECLARE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/setMuted.h:34]:
```suggestion
};

```

File: extensions/src/ACRE2Core/setPTTKeys.h [hunk 13]
```diff
@@ -16,7 +16,7 @@ RPC_FUNCTION(setPTTKeys) {
 
         */
 
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 DECLARE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/setPTTKeys.h:22]:
```suggestion
};

```

File: extensions/src/ACRE2Core/setSelectableVoiceCurve.h [hunk 14]
```diff
@@ -31,7 +31,7 @@ RPC_FUNCTION(setSelectableVoiceCurve) {
         UNLOCK(CEngine::getInstance()->getSelf());
     }
 
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 DECLARE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/setSelectableVoiceCurve.h]:
```suggestion
};

```

File: extensions/src/ACRE2Core/setSoundSystemMasterOverride.h [hunk 15]
```diff
@@ -24,7 +24,7 @@ RPC_FUNCTION(setSoundSystemMasterOverride) {
     }
 
 
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 DECLARE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/setSoundSystemMasterOverride.h]:
```suggestion
};

```

File: extensions/src/ACRE2Core/stopRadioSpeaking.h [hunk 16]
```diff
@@ -15,9 +15,9 @@
...
 RPC_FUNCTION(stopRadioSpeaking) {
     
-    CEngine::getInstance()->getClient()->localStopSpeaking(acre_speaking_radio);
+    CEngine::getInstance()->getClient()->localStopSpeaking(AcreSpeaking::radio);
 
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 DECLARE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/stopRadioSpeaking.h]:
```suggestion
};

```

File: extensions/src/ACRE2Core/updateSelf.h [hunk 17]
```diff
@@ -33,7 +33,7 @@ RPC_FUNCTION(updateSelf) {
     CEngine::getInstance()->getSelf()->setCurrentLanguageId(vMessage->getParameterAsInt(6));
 
     UNLOCK(CEngine::getInstance()->getSelf());
-    return acre_result_ok;
+    return AcreResult::ok;
 }
 DECLARE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Core/updateSelf.h]:
```suggestion
};

```

File: extensions/src/ACRE2Shared/IRpcFunction.h [hunk 18]
```diff
@@ -9,7 +9,7 @@ class IRpcFunction
 public:
     virtual ~IRpcFunction(){}
 
-    virtual acre_result_t call(IServer*, IMessage *) = 0;
+    virtual AcreResult call(IServer*, IMessage *) = 0;
 
     DECLARE_INTERFACE_MEMBER(char *, Name);
 };
```
Comment (jonpas) [inline extensions/src/ACRE2Shared/IRpcFunction.h]:
```suggestion
};

```

**Review by jonpas — CHANGES_REQUESTED** [review]
Does this fit our guidelines? I am not sure what @NouberNou thinks of enum namings.

EOF unrelated, but might as well since some did get cleaned.

Looking good otherwise.

**Comment by TheMagnetar** [comment]
Leave those spaces. They are all over the code due to the use of macros.

**Comment by jonpas** [comment]
What spaces? I didn't remove them before closing braces, I just added a newline at EOF. No clue what GitHub did, it doesn't even show a space before it in the diff though.

**Comment by jonpas** [comment]
Still some left? (I resolved those committed.)

**Review by jonpas — APPROVED** [review]

**Pull request merged**